### PR TITLE
add snapml solver

### LINKED ISSUE #83: SOLVER add snapML?

### PR DESCRIPTION
**Pull request by tanglef**
Fixes #83 by adding snapml solver

This solver is not good when intercept is True or with small datasets, but for example with rcv1:
One example of run 
Capture d’écran de 2022-05-09 18-27-53

### CODE REVIEW

**Comment by tomMoral** [comment]
nice!!!

File: test_config.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import sys
-
+from benchopt.utils.sys_info import _get_cuda_version
 import pytest
```
Comment (tomMoral) [inline test_config.py:3]: Put `benchopt` related import at the end of the imports (from more general stdlib to particular packages)

```suggestion
import pytest

from benchopt.utils.sys_info import _get_cuda_version
```

File: solvers/snapml.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+class Solver(BaseSolver):
+    name = "snapml"
+
+    install_cmd = "conda"
+    requirements = ["pip:snapml"]
+
+    parameters = {"gpu": [True, False]}
+
+    def skip(self, X, y, lmbd):
```
Comment (tomMoral) [inline solvers/snapml.py:18]:
```suggestion
    def skip(self, X, y, lmbd, fit_intercept):
```

**Review by tomMoral — APPROVED** [review]
LGTM

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
thx @tanglef !

**Pull request merged**